pkg/controllers/disruption/consolidation.go

```diff
@@ -192,6 +193,52 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 		results.NewNodeClaims[0].Requirements.Add(scheduling.NewRequirement(v1beta1.CapacityTypeLabelKey, v1.NodeSelectorOpIn, v1beta1.CapacityTypeSpot))
```

Comment:
musing: Thinking on this even more, I'm wondering if we want to reason in terms of instance type flexibility when we are going from on-demand -> spot as well. From a high-level abstraction perspective, is the thing that we are reasoning about in both spot -> spot and on-demand -> spot any different? At what point do you want to consolidate to a spot node from an on-demand node just strictly based on whether the instance type is cheaper? Do we need the same flexibility constraints to ensure that availability is maintained? I'm fine punting on this and not addressing this in this PR but mainly just thinking if it might be worth trying to collapse the number of concepts that we have here.

cc: @njtran